node/subsystem-util/src/lib.rs

```diff
@@ -205,6 +207,49 @@ specialize_requests! {
 	fn request_on_chain_votes() -> Option<ScrapedOnChainVotes>; FetchOnChainVotes;
 }
 
+/// Requests session info from the runtime effective at given relay-parent. First obtains
+/// session index at the relay-parent, relying on the fact that it should be cached by the runtime
+/// API caching layer even if the block itself has already been pruned. Then requests session info
+/// by session index.
+/// Returns an error if failed to communicate to the runtime, or the session info is not in the
+/// storage, which should never happen.
+pub async fn session_info_at_relay_parent(
```

Comment:
Consider using [this](https://github.com/paritytech/polkadot/blob/03c48861f73c30402795c89634c0cc343e6f0fe1/node/subsystem-util/src/runtime/mod.rs#L146) already existing functionality. Example usage [here](https://github.com/paritytech/polkadot/blob/702d4fc7d502ae7cbfa0a4011101129c5bb5fff6/node/network/availability-distribution/src/lib.rs#L56).

Reply:
Yeah, I saw it. However, I decided not to use it because it fetches `ExtendedSessionInfo`, doing some extra work we don't need (not much of it, but still...)

Feel free to persuade me to consider the overhead negligible. This function was used in 3 places before, and now it's just a single place left, so probably it's a good idea to get rid of it.

Reply:
It does slightly more work (which should indeed be negligible, at the same time it caches results. So the minute you fetch the same `SessionInfo` or the same `SessionIndex` twice it will be faster.